### ISSUE TITLE
Make `SIGHUP` reload and update cache factor values 

### ISSUE THREAD

**Issue by erikjohnston** [matrix-org/synapse#12373]
This would be *very* useful when you want to update cache factors across many workers without having to restart them all (e.g. when they're under load and restarting would cause issues)

**Comment by DMRobertson** [comment]
Is this as simple as having a SIGHUP handler call this function?

https://github.com/matrix-org/synapse/blob/4586119f0b0901be64f08655d3aaaef289a51bde/synapse/config/cache.py#L264-L274

**Comment by DMRobertson** [comment]
Oh no, we'd need to get the homeserver to reparse the config first.

**Comment by clokep** [comment]
We do something similar for logging config near https://github.com/matrix-org/synapse/blob/8d156ec0ba17d848581f18aa40ebfd76dda763d4/synapse/config/logger.py#L345

**Comment by DMRobertson** [comment]
That's a helpful pointer, thanks. WIll still have to work out how to wrangle our config class magic.

**Comment by clokep** [comment]
Yeah, although I realize now it isn't quite the same since the logging stuff has a separate config file.
